src/lib/operators/abstract_operator.cpp

```diff
@@ -67,6 +67,30 @@ void AbstractOperator::execute() {
                 _output ? _output->row_count() : 0, _output ? _output->chunk_count() : 0,
                 reinterpret_cast<uintptr_t>(this));
 
+  // Chunks from operators other than GetTable should be immutable
+  if constexpr (HYRISE_DEBUG) {
+    // We exclude the TableWrapper (heavily used in tests with arbitrary tables) and modifiying operators.
+    if (std::dynamic_pointer_cast<const AbstractReadOnlyOperator>(_input_left) &&
+        _input_left->type() != OperatorType::TableWrapper) {
+      const auto check_input_table = [&](const std::shared_ptr<const Table>& input_table) {
+        const auto chunk_count = input_table->chunk_count();
+        for (auto chunk_id = ChunkID{0u}; chunk_id < chunk_count; ++chunk_id) {
+          const auto chunk = input_table->get_chunk(chunk_id);
```

Comment:
```suggestion
          const auto& chunk = input_table->get_chunk(chunk_id);
```

Reply:
That wouldn't work with the following assert as references cannot be null. `get_chunk()` however can return a nullptr (see `remove_chunk()`).

Reply:
Nah, it would be a reference to a shared_ptr, which can be null. For data tables, we would have a lifetime extension, but for reference tables, we could save a shared_ptr copy

Reply:
Ah I see. Then again: is there a reason we do it differently pretty much everywhere? Should the operator really be part of ensuring that a chunk is not unwillingly freed?

Reply:
Chunks should not suddenly disappear, so it probably is a small inconsistency.

Reply:
Do we a common understanding/guide which component ensure that chunks are kept alive during execution of an operator?